WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingTaskView.swift

```diff
@@ -0,0 +1,130 @@
+import SwiftUI
+
+/// Shows a tappable onboarding task to set up the store. If the task is complete, a checkmark is shown.
+struct StoreOnboardingTaskView: View {
+    private let viewModel: StoreOnboardingViewModel.TaskViewModel
+    private let onTap: (StoreOnboardingTask) -> Void
+
+    init(viewModel: StoreOnboardingViewModel.TaskViewModel,
+         onTap: @escaping (StoreOnboardingTask) -> Void) {
+        self.viewModel = viewModel
+        self.onTap = onTap
+    }
+
+    /// Scale of the view based on accessibility changes.
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    var body: some View {
+        Button {
+            onTap(viewModel.task)
+        } label: {
+            HStack(alignment: .center, spacing: Layout.horizontalSpacing) {
+                // Check icon or task icon.
+                if viewModel.isComplete {
+                    Image(uiImage: .checkCircleImage.withRenderingMode(.alwaysTemplate))
+                        .resizable()
+                        .foregroundColor(.init(uiColor: .accent))
+                        .frame(width: scale * Layout.imageDimension,
+                               height: scale * Layout.imageDimension)
+                } else {
+                    Image(uiImage: viewModel.icon.withRenderingMode(.alwaysTemplate))
+                        .resizable()
+                        .foregroundColor(.init(uiColor: .text))
+                        .frame(width: scale * Layout.imageDimension,
+                               height: scale * Layout.imageDimension)
+                }
```

Comment:
💡 Nit: this can be simplified with conditional checks - you can also move the checks to the view model if needed:

```suggestion
                Image(uiImage: viewModel.isComplete ? .checkCircleImage : viewModel.icon)
                    .renderingMode(.template)
                    .resizable()
                    .foregroundColor(.init(uiColor: viewModel.isComplete ? .accent : .text))
                    .frame(width: scale * Layout.imageDimension,
                           height: scale * Layout.imageDimension)
```